crates/engine/src/io.rs

```diff
@@ -38,6 +40,63 @@ pub trait OutputBuffers {
     fn stderr(&self) -> &[u8];
 }
 
+/// Wrapper around File w/ a convienient PathBuf for cloning
+pub struct PipeFile(pub(crate) File, pub(crate)PathBuf);
+
+impl PipeFile {
+    /// Constructs an instance from a set of PipeFile objects.
+    pub fn new(
+        file: File,
+        path: PathBuf,
+    ) -> Self {
+        Self(
+            file,
+            path,
+        )
+    } 
+}
+
+impl std::fmt::Debug for PipeFile {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PipeFile")
+            .field("File", &self.0)
+            .field("PathBuf", &self.1)
+            .finish()
+    }
+}
+
+impl Clone for PipeFile {
+    fn clone(&self) -> Self {
+        let f = OpenOptions::new()
+                    .read(true)
+                    .write(true)
+                    .open(&self.1)
+                    .unwrap();
+        Self(f, self.1.clone())
+    }
+}
+
+/// CustomIoPipes that can be passed to `ExecutionContextConfiguration`
+/// to direct out and err
+#[derive(Clone, Debug)]
+pub struct CustomLogPipes {
+    pub(crate) stdout_pipe: PipeFile,
```

Comment:
what's the reason for leaving stdin_pipe behind?

Reply:
I've just added it!